### ISSUE TITLE
Explicit Spartan AntiCheat support

### ISSUE THREAD

**Issue by josegoust** [kernitus/BukkitOldCombatMechanics#764]
## Proposed solution
Add the spartan api back
https://github.com/Vagdedes/Spartan-AntiCheat

U removed the spartan api, pls add again

**Comment by kernitus** [comment]
Is there a module/feature in particular that you find is affected by Spartan? As of our latest tests Spartan works fine with OCM without explicit support.

**Comment by Vagdedes** [comment]
You were using the API previously, just add back that code with the new repository as the dependency. There are actually false positives with the movement checks sometimes, my reports increased since your changes. The other repository you rightfully removed but it should have been a replacement, not just a deletion.

Thanks

**Comment by kernitus** [comment]
> You were using the API previously, just add back that code with the new repository as the dependency. There are actually false positives with the movement checks sometimes, my reports increased since your changes. The other repository you rightfully removed but it should have been a replacement, not just a deletion.
> 
> Thanks

The Spartan API was only ever used to check for healing ticks (291c356) in the old player regen module, so false positives about movement checks are completely unrelated to its removal. This is the first we are hearing of reports relating to movement - and adding the dependency back in as-is would make no difference. The only places where we might affect movement is in the modules that modify combat knockback, such as the [FishingRodVelocity](https://github.com/kernitus/BukkitOldCombatMechanics/blob/master/src/main/java/kernitus/plugin/OldCombatMechanics/module/ModuleFishingRodVelocity.java), [PlayerKnockback](https://github.com/kernitus/BukkitOldCombatMechanics/blob/master/src/main/java/kernitus/plugin/OldCombatMechanics/module/ModulePlayerKnockback.java), and [ProjectileKnockback](https://github.com/kernitus/BukkitOldCombatMechanics/blob/master/src/main/java/kernitus/plugin/OldCombatMechanics/module/ModuleProjectileKnockback.java) modules. Have you any suggestions on which checks might need to be disabled and when?

**Comment by Vagdedes** [comment]
I'd definitely recommend disabling checks momentarily via the cancelCheck() API method for up to 100 ticks when an OCM feature is triggered, any of it. Thanks for the detailed feedback on the commits, it's probably our changes that caused this. Let me know if something can be done

**Comment by kernitus** [comment]
> I'd definitely recommend disabling checks momentarily via the cancelCheck() API method for up to 100 ticks when an OCM feature is triggered, any of it. Thanks for the detailed feedback on the commits, it's probably our changes that caused this. Let me know if something can be done

OCM does calculations on every EntityDamageByEntity event, wouldn't doing that effectively disable all combat anticheats from running?

**Comment by Vagdedes** [comment]
Yes, that's true, although I was referring more to movement checks. Either way, the co-developer said they would look at it since they are the ones who made the recent detections responsible for the false positives.
